### PR TITLE
[12.0][T968][ADD]- Added Sale Order Pricelist commitment date module

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
- Task #[968](https://www.quartile.co/web#id=968&action=771&model=project.task&view_type=form&menu_id=505)
 - Added `C901` in flake 8 file for ignore this warning
`sale_order_pricelist_commitment_date/models/product_pricelist.py:14:5: C901 'Pricelist._compute_price_rule' is too complex (33)`

### CODE REVIEW

File: sale_order_pricelist_commitment_date/models/product_pricelist.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+from odoo.exceptions import UserError
+
+
+class Pricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    # This method overide due to product pricelist will be calculated based
+    # on sale commitment date
+    @api.multi
+    def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
```
Comment (TimLai125) [inline sale_order_pricelist_commitment_date/models/product_pricelist.py:11]:
```
@api.multi
def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
    commitment_date = self._context.get("commitment_date") or False
    commitment_date = commitment_date or self._context.get("date") or fields.Date.today()
    super(Pricelist, self)._compute_price_rule(products_qty_partner, date=commitment_date, uom_id=uom_id)
```

**Comment by yostashiro** [comment]
@YogeshMahera-SerpentCS I just checked the behavior myself. Maybe the approach we should take is to pass commitment_date to the context for all the line fields that triggers price updates (i.e. `product_id`, `product_uom_qty`, `product_uom`, `secondary_uom_qty`, `secondary_uom_id`).

Besides above, if we trigger price update when commitment_date is changed, we should cover all the possible patterns, I suppose.  What do you think?

**Comment by YogeshMahera-SerpentCS** [comment]
Hi, @yostashiro I have amended changes for product_uom_qty,product_uom for form, and tree view both, Also amended changes while commitment_date change the already created order line will get the updated commitment_date unit price.

for `secondary_uom_qty,secondary_uom_id` I did check the when I am going to change on this field it's not going to change on-call this method `compute_price_rule`

So even If I pass the context on this field this method does not have any change in unit price. Let me know if I miss something else.

**Comment by YogeshMahera-SerpentCS** [comment]
Hi @yostashiro Can you please check I have passed the context to `secondary_uom_id`

File: sale_order_pricelist_commitment_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    "name": "Sale Order Pricelist Commitment Date",
+    "summary": "",
+    "version": "12.0.1.0.0",
+    "category": "Sales",
+    "website": "https://www.quartile.co/",
+    "author": "Quartile Limited",
+    "license": "LGPL-3",
+    "application": False,
```
Comment (yostashiro) [inline sale_order_pricelist_commitment_date/__manifest__.py]:
```suggestion
```

File: sale_order_pricelist_commitment_date/tests/test_sale_pricelist.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            }
+        )
+        self.commitment_date = date.today() + relativedelta(days=+10)
+        self.sale_order.write({"commitment_date": self.commitment_date})
+        self.sale_order.onchange_commitment_date()
+        self.assertEquals(
+            fields.Date.to_date(self.sale_order.commitment_date),
+            self.commitment_date,
+            "In onchange method the commitment date pass date through context",
+        )
```
Comment (yostashiro) [inline sale_order_pricelist_commitment_date/tests/test_sale_pricelist.py]: Can't tell what this test is trying to prove...

File: sale_order_pricelist_commitment_date/__manifest__.py [hunk 4]
```diff
@@ -10,5 +10,5 @@
     "license": "LGPL-3",
     "installable": True,
     "depends": ["sale_order_secondary_unit"],
-    "data": ["views/sale_view.xml"],
+    "data": ["views/sale_order_view.xml"],
```
Comment (yostashiro) [inline sale_order_pricelist_commitment_date/__manifest__.py]:
```suggestion
    "data": ["views/sale_order_views.xml"],
```

Following the OCA convention: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#file-naming

**Review by yostashiro — CHANGES_REQUESTED** [review]
@YogeshMahera-SerpentCS Just a minor point. Please fix this and merge the PR.

**Comment by YogeshMahera-SerpentCS** [comment]
Hi, @yostashiro I have amended changes but it seems I don't have access to merge.

**Comment by yostashiro** [comment]
@YogeshMahera-SerpentCS Try again please.

**Pull request merged**